docs/en/ingest-management/fleet-agent-proxy-support.asciidoc

```diff
@@ -0,0 +1,310 @@
+* Send data to {es}
+* Retrieve agent policies from {fleet-server}
+* Retrieve agent policies from {es} (only needed for agents running {fleet-server})
+* Monitor services (not shown)
+
+image::images/agent-proxy-server.png[Image showing connections between {agent}, {fleet-server}, and {es}]
+
+You can configure {agent} proxy settings by specifying:
+
+* <<host-proxy-env-vars>>
+* <<proxy-settings-in-agent-policy>>
+* <<cli-proxy-settings>>
+
+[discrete]
+[[host-proxy-env-vars]]
+=== Host environment variables
+
+Set environment variables on the host to configure default proxy settings.
+The {agent} uses host environment settings by default if no proxy settings are
+specified elsewhere. You can override host proxy settings later when you
+configure the {agent} and {fleet} settings. The following environment variables
+are available on the host:
+
+|===
+|Variable |Description
+
+|`HTTP_PROXY`
+|URL of the proxy server for HTTP traffic. 
+
+|`HTTPS_PROXY`
+|URL of the proxy server for HTTPS traffic.
+
+|`NO_PROXY`
+|IP addresses or domain names that should not use the proxy. Supports patterns.
+|===
+
+The proxy URL can be a complete URL or `host[:port]`, in which case the `http`
+scheme is assumed. An error is returned if the value is a different form.
+
+[discrete]
+[[where-to-set-proxy-env-vars]]
+==== Where to set proxy environment variables
+
+The location where you set these environment variables is platform-specific and
+based on the system manager you're using. Here are some examples to get you
+started. For more information about setting environment variables, refer to the
+documentation for your operating system.
+
+* For Windows services, set environment variables for the service in
+the Windows registry.
++
+This PowerShell command sets the
+`HKLM\SYSTEM\CurrentControlSet\Services\Elastic Agent\Environment` registry
+key, then restarts {agent}:
++
+[source,shell]
+----
+$environment = [string[]]@(
+  "HTTPS_PROXY=https://proxy-hostname:proxy-port",
+  "HTTP_PROXY=http://proxy-hostname:proxy-port"
+  )
+
+Set-ItemProperty "HKLM:SYSTEM\CurrentControlSet\Services\Elastic Agent" -Name Environment -Value $environment
+
+Restart-Service "Elastic Agent"
+----
+
+* For Linux services, the location depends on the distribution you're using.
+For example, you can set environment variables in:
+
+** `/etc/sysconfig/elastic-agent` for Red Hat-like distributions
+
+** `/etc/default/elastic-agent` for Debian and Ubuntu distributions
++
+For example:
++
+[source,shell]
+----
+HTTPS_PROXY=https://my.proxy:8443
+HTTP_PROXY=http://my.proxy:8080
+----
+
+** `/etc/systemd/system/elastic-agent.service` for systems that use `systemd` to
+manage the service. To edit the file, run:
++
+[source,shell]
+----
+sudo systemctl edit --full elastic-agent.service
+----
++
+Then add the environment variables under `[Service]`
++
+[source,shell]
+----
+[Service]
+
+Environment="HTTPS_PROXY=https://my.proxy:8443"
+Environment="HTTP_PROXY=http://my.proxy:8080"
+----
```

Comment:
(nitpick) I'de move this chunk before the explanation of `/etc/default` and `/etc/sysconfig` as I think more systems use systemd

Reply:
@michel-laterman Recent versions of ubuntu also use systemd, so should an ubuntu user put them in `[Service]` or in `/etc/default/elastic-agent`?

Reply:
I think using systemd syntax as our first documented option is better as it's applicable to rhel/debian based instances.
I believe Our service files for .deb and .rpm installers should have (optional) references to the distro specific files (like `/etc/default/elastic-agent`)